woocommerce/woocommerce-android
# Onboarding: Country selector search

Part of: https://github.com/woocommerce/woocommerce-android/issues/9702


### Description


This PR adds search functionality on the country list picker. See demo video below:

[country-picker-search-demo.webm](https://github.com/woocommerce/woocommerce-android/assets/266376/1893341c-8aa0-411a-80e5-859bb33bae28)


### Testing instructions


1. Start store creation,
2. Go through onboarding until the "Where is your business located?" screen is shown, 
3. Tap the current country, ensure the country picker selector is shown
4. Ensure search bar at the top is shown,
5. Try searching for various countries and ensure things work well,
6. Pick a country from the search result and ensure you're taken back to the "Where is your business located?" screen.
